### PR TITLE
[TRTLLM-13696][test] Part2.1: Migrate CPU only tests - runtime

### PR DESCRIPTION
**Pull request by tongyuantongyu**
## Dev Engineer Review

- Updated CI pytest argument construction so `--unittest-markexpr` for stages whose `stageName` starts with `CPU-` now targets `cpu_only` (instead of `cpu_only and not disabled`); non-`CPU-*` stages still use `not cpu_only`.
- Adjusted CPU stage mappings in `jenkins/L0_Test.groovy` so `CPU-Generic-x86-1` and `CPU-Generic-arm-1` use the `l0_cpu` test-db context (replacing `l0_cpu_x86` / `l0_cpu_arm`).
- Fixed `test_unittests_v2` mark-expression handling in `tests/integration/defs/test_unittests.py`:
  - If a per-case `-m <expr>` already exists, it is rewritten to `(<case_expr>) and (<unittest_markexpr>)`.
  - Otherwise, `-m <unittest_markexpr>` is appended to the per-case pytest `arg_list`.
- Removed CUDA compute-capability-dependent `dtype` normalization/validation from `BaseLlmArgs` in `tensorrt_llm/llmapi/llm_args.py` by deleting the `validate_dtype` validator that previously rewrote `dtype="auto"`→`"float16"` on pre-SM80 CUDA and rejected `dtype="bfloat16"` on those devices.
- Migrated the first runtime/unit-test wave to CPU-only execution via `pytest.mark.cpu_only` (mostly module-level `pytestmark`, plus a few decorator-level marks) and small determinism/compatibility tweaks where needed (e.g., async URL fetching switched to `asyncio.run()`, MPI/tiktoken cache fixtures added/adjusted for CPU determinism).
- Updated `tests/README.md` to clarify CPU-only CI guidance and the GPU-independent unit-test marking approach, and to improve the CI-bridge documentation (including updated wording and examples).

Key risk areas to review:
- Verify that every `cpu_only`-marked test genuinely avoids CUDA/GPU-only code paths and doesn’t rely on prior GPU-dependent `dtype` rewrite/rejection behavior.
- Confirm the new `l0_cpu` tier plus the `CPU-Generic-*`→`l0_cpu` mapping fully covers the intended runtime/unit migration.
- Validate that removing SM80/pre-SM80 `dtype` auto-rewrite/rejection doesn’t cause failures in tests expecting the old behavior.

### Test-db / CI list changes (test-list-only scope)
- Modified/added/removed files under `tests/integration/test_lists/test-db/`:
  - Added: `l0_cpu.yml` (new `system_gpu_count: 0` / `cpu_only` tier; `stage: pre_merge`, `backend: generic`, `orchestrator: mpi`).
  - Removed: `l0_cpu_arm.yml` and `l0_cpu_x86.yml`.
  - Updated selections:
    - `l0_a10.yml` (reduced executor/llmapi module sets to match CPU-only migration targets).
    - `l0_a100.yml` (removed `unittest/llmapi/test_executor.py`; split `unittest/llmapi/test_llm.py` into part0 vs not-part0 runs).
    - `l0_b200.yml` (removed specific tests from KV-cache stats and post-merge backend pytorch lists).
    - `l0_dgx_h100.yml` (swap/adjust selected distributed and llmapi tests).
    - `l0_gh200.yml` (update backend pytorch entry from `test_llm_quant.py` to `unittest/bindings`).
    - `l0_h100.yml` (swap/remove a few Ray and post-merge PyTorch entries).
- Verdict: needs follow-up (CBTS/coverage data and final migration run outcomes weren’t provided here; earlier CI comments indicate iterative failures before later partial success).

### Test code changes (outside `test_lists/`)
Touched test logic/selection mechanics and key determinism points:
- `tests/integration/defs/test_unittests.py`
  - Modified: `test_unittests_v2` to merge `-m` mark expressions correctly into per-case `arg_list`.
  - Coverage in test-db: intended to be exercised by the updated CPU stage/test-db targeting; needs follow-up to confirm full coverage.
- `tests/unittest/llmapi/test_mpi_session.py`
  - Modified: added an autouse fixture to enable MPI (removes `TLLM_DISABLE_MPI`), and broadened CPU-only gating for specific tests.
  - Coverage in test-db: intended to be covered by the new CPU tier; needs follow-up.
- `tests/unittest/inputs/test_url_validation.py`
  - Modified: switched asyncio coroutine driving to `asyncio.run()` and marked module CPU-only.
  - Coverage in test-db: intended to be covered by CPU-only stage selections; needs follow-up.
- `tests/unittest/_torch/modules/dwdp/test_dwdp_mapping.py`
  - Modified: marked CPU-only and added an autouse fixture to patch `tensorrt_llm.mapping.mpi_disabled` for consistent behavior under CPU.
  - Coverage in test-db: intended to be covered by CPU tier; needs follow-up.
- `tests/unittest/llmapi/test_kv_cache_dtype_override.py`
  - Modified: adjusted `get_llm_args(...)` invocation to pass `gpus_per_node=1` alongside `kv_cache_dtype="nvfp4"` and marked CPU-only.
  - Coverage in test-db: intended to be covered by CPU-only selections; needs follow-up.

Widespread marking-only changes:
- Many modules under `tests/unittest/**` now set `pytestmark = pytest.mark.cpu_only` (module-level or per-test decorators), restricting execution to CPU environments without otherwise changing assertions.

- Verdict (test code): needs follow-up (large CPU-only test surface expanded; requires confirming CI stability across the new `l0_cpu` tier and the `CPU-Generic-*` mappings).

## Description

First wave of tests that can be moved to run without GPU.

## Test Coverage

The 2 CPU pipelines.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

**Comment by tongyuantongyu** [comment]
/bot run --stage-list "CPU-Generic-x86-1, CPU-Generic-arm-1" --disable-fail-fast

**Comment by tongyuantongyu** [comment]
/bot run --stage-list "CPU-Generic-x86-1, CPU-Generic-arm-1" --disable-fail-fast

**Comment by tongyuantongyu** [comment]
/bot run --stage-list "CPU-Generic-x86-1, CPU-Generic-arm-1" --disable-fail-fast

**Comment by tongyuantongyu** [comment]
/bot run --stage-list "CPU-Generic-x86-1, CPU-Generic-arm-1" --disable-fail-fast

**Comment by tongyuantongyu** [comment]
/bot run --stage-list "CPU-Generic-x86-1, CPU-Generic-arm-1" --disable-fail-fast

File: tests/unittest/llmapi/test_tokenizer_multinode.py [hunk 1]
```diff
@@ -9,6 +9,8 @@
 
 from tensorrt_llm.tokenizer.tokenizer import TransformersTokenizer, load_hf_tokenizer
 
+pytestmark = pytest.mark.cpu_only
+
```
[the remark this answers is not in the document]
Comment (tongyuantongyu) [reply tests/unittest/llmapi/test_tokenizer_multinode.py:13]: This PR is a migration of pre-existing tests. This file (and some others) are not actively tested on CI currently, so as after this PR. They will be addressed in the future instead, to reduce the chance they fail CI and block this PR.

This message only serves as a note for reviewers to understand the situation. No need to reply.

**Review by EmmaQiaoCh — APPROVED** [review]

File: tests/integration/test_lists/test-db/l0_cpu_arm.yml
Comment (EmmaQiaoCh) [inline tests/integration/test_lists/test-db/l0_cpu_arm.yml]: Maybe we can use this filter instead of 2 duplicate yml for x86 and aarch64.

**Review by QiJune — APPROVED** [review]

**Comment by tongyuantongyu** [comment]
/bot run --stage-list "CPU-Generic-x86-1, CPU-Generic-arm-1" --disable-fail-fast

**Comment by tongyuantongyu** [comment]
/bot run --stage-list "CPU-Generic-x86-1, CPU-Generic-arm-1" --disable-fail-fast

**Review by StanleySun639 — APPROVED** [review]

**Comment by tongyuantongyu** [comment]
/bot run --stage-list "CPU-Generic-x86-1, CPU-Generic-arm-1" --disable-fail-fast

File: tests/unittest/_torch/lora/test_moe_lora_extract.py [hunk 2]
```diff
@@ -17,6 +17,9 @@
 import pytest
 import torch
 
+pytestmark = pytest.mark.cpu_only
+
+
```
[the remark this answers is not in the document]
Comment (tongyuantongyu) [reply tests/unittest/_torch/lora/test_moe_lora_extract.py:22]: This PR only **moves** active tests to CPU stage. If they are not currently running in any stage, we are not going to make them run on CPU stage in this PR. Please confirm this.

Marks are added because we scanned all tests, but to avoid unexpected CI fail we don't enable them now in this PR.

**Comment by BowenFu** [comment]
The test migration itself checks out — I verified the `l0_cpu.yml` union covers both deleted per-arch lists, the `-m` merge in `test_unittests.py` is correct for every current test-db expression (and actually fixes the old two-separate-`-m` override), and no removed GPU entry loses coverage.

One thing to fix before this lands: deleting `BaseLlmArgs.validate_dtype` also deletes behavior the field still documents. `dtype`'s own `description` (`llm_args.py:4252`) says *"...and is overridden to float16 on GPUs with compute capability < 8.0 (pre-Ampere)"*, and nothing else in the PR or in the model-loading path restores it — `_torch/model_config.py` resolves `auto` from the checkpoint without consulting compute capability. So on pre-SM80 `dtype="auto"` can now land on BF16 and fail deep in kernel init instead of selecting FP16, and `dtype="bfloat16"` loses its clear configuration-time error.

SM70/75 are no longer supported build targets, so this likely affects nobody in practice — but the description is user-facing and it now promises something the code doesn't do. Guarding instead of removing keeps CPU-only construction working:

```python
if (torch.cuda.is_available()
        and torch.cuda.get_device_properties(0).major < 8):
```

If the intent is genuinely to drop pre-Ampere handling, that's fine too — but then it belongs in its own PR with the field description updated, not folded into an 84-file test migration.

Minor: a comment in `test_multi_frontend_routing.py` still refers to the old `cpu_only and not disabled` expression.
